### PR TITLE
fix(swift): Data - Set up Amplify Data code improvements

### PR DESCRIPTION
**Pull request by lawmicha**
#### Description of changes:
This PR updates the Swift / Data / "Set up Amplify Data" page and "Connect your app code" to API page

### Instructions

**If this PR should not be merged upon approval for any reason, please submit as a DRAFT**

Which product(s) are affected by this PR (if applicable)?
- [ ] amplify-cli
- [ ] amplify-ui
- [ ] amplify-studio
- [ ] amplify-hosting
- [x] amplify-libraries

Which platform(s) are affected by this PR (if applicable)?
- [ ] JS
- [x] Swift
- [ ] Android
- [ ] Flutter
- [ ] React Native

**Please add the product(s)/platform(s) affected to the PR title**

#### Checks

- [ ] Does this PR conform to [the styleguide](https://github.com/aws-amplify/docs/blob/main/STYLEGUIDE.md)?

- [ ] Does this PR include filetypes other than markdown or images? Please add or update unit tests accordingly.

- [ ] Are any files being deleted with this PR? If so, have the needed redirects been created?

- [ ] Are all links in MDX files using the MDX link syntax rather than HTML link syntax? <br /> 
      _ref: MDX: `[link](https://docs.amplify.aws/)` 
            HTML: `<a href="https://docs.amplify.aws/">link</a>`_

### When this PR is ready to merge, please check the box below
- [ ] Ready to merge

_By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license._

### CODE REVIEW

**Review by renebrandel — DISMISSED** [review]

File: src/pages/[platform]/build-a-backend/data/set-up-data/index.mdx [hunk 1]
```diff
@@ -444,20 +442,53 @@ com.example.MyAmplifyApp I/MyAmplifyApp: Added Todo with id: SOME_TODO_ID
...
+
     var body: some View {
         // highlight-start
         VStack {
             Button(action: {
-                addNewTodo()
+                vm.createTodo()
             }) {
                 HStack {
                     Text("Add a New Todo")
```
Comment (lawmicha) [inline src/pages/[platform]/build-a-backend/data/set-up-data/index.mdx:494]: image

File: src/pages/[platform]/build-a-backend/data/set-up-data/index.mdx [hunk 2]
```diff
@@ -648,57 +660,66 @@ If you build and rerun the application, you should see the todo that was created
...
-            print("Successfully retrieved list of todos: \(todos)")
-        case .failure(let error):
-            print("Got failed result with \(error.localizedDescription)")
-        }
-        } catch {
-            print("Failed to query list of todos: \(error)")
-        }
-    }
-    // highlight-end
 }
```
Comment (lawmicha) [inline src/pages/[platform]/build-a-backend/data/set-up-data/index.mdx:723]: image

File: src/pages/[platform]/build-a-backend/data/set-up-data/index.mdx [hunk 3]
```diff
@@ -934,68 +955,33 @@ setContent {
...
                 print("Subscription has terminated with \(error)")
             }
-            // highlight-end
         }
     }
 
-    // highlight-start
     private func handleSubscriptionEvent(_ subscriptionEvent: GraphQLSubscriptionEvent<Todo>) {
         switch subscriptionEvent {
         case .connection(let subscriptionConnectionState):
```
Comment (lawmicha) [inline src/pages/[platform]/build-a-backend/data/set-up-data/index.mdx:987]: image

File: src/pages/[platform]/build-a-backend/data/set-up-data/index.mdx [hunk 4]
```diff
@@ -1010,8 +996,47 @@ struct ContentView: View {
...
+        // highlight-start
+        .onDisappear {
+            vm.cancel()
+        }
+        .task {
+            vm.listTodos()
+            vm.subscribe()
+        }
+        // highlight-end
+    }
```
Comment (lawmicha) [inline src/pages/[platform]/build-a-backend/data/set-up-data/index.mdx:1039]: image

File: src/pages/[platform]/build-a-backend/data/set-up-data/index.mdx [hunk 5]
```diff
@@ -1010,8 +996,47 @@ struct ContentView: View {
...
+```
+
+Then in `ContentView.swift`, when the view appears, call `vm.subscribe()`. On disappear, cancel the subscription.
+
+
+```swift title="ContentView.swift"
+
+import SwiftUI
+import Amplify
+
```
Comment (lawmicha) [inline src/pages/[platform]/build-a-backend/data/set-up-data/index.mdx:1022]:
```suggestion
```

File: src/pages/[platform]/build-a-backend/data/set-up-data/index.mdx (hunk 4 above)
Comment (lawmicha) [reply src/pages/[platform]/build-a-backend/data/set-up-data/index.mdx]: removed imports

**Review by renebrandel — APPROVED** [review]

**Review by renebrandel — DISMISSED** [review]

File: src/pages/[platform]/build-a-backend/data/set-up-data/index.mdx [hunk 6]
```diff
@@ -444,20 +442,53 @@ com.example.MyAmplifyApp I/MyAmplifyApp: Added Todo with id: SOME_TODO_ID
...
+Create a new file called `TodoViewModel.swift` and the `createTodo` function with the following code:
 
-```swift title="ContentView.swift"
-import SwiftUI
-// highlight-next-line
+```swift title="TodoViewModel.swift"
 import Amplify
 
+@MainActor
+class TodoViewModel: ObservableObject {
```
Comment (ruisebas) [inline src/pages/[platform]/build-a-backend/data/set-up-data/index.mdx:451]: You need to import `Foundation` for `ObservableObject` to be accessible
image

File: src/pages/[platform]/build-a-backend/data/set-up-data/index.mdx [hunk 7]
```diff
@@ -934,68 +955,33 @@ setContent {
 
 To add real-time updates, you can use the subscription feature of Amplify Data. It allows to subscribe to `onCreate`, `onUpdate`, and `onDelete` events of the application. In our example, let's append the list every time a new todo is added.
 
-First, add a private variable to store the subscription. Then, on `init()` establish the subscription and on disappear, cancel the subscription.
+First, add a private variable to store the subscription. Then on the `init()` create the subscription, and add the `subscribe()` and `cancel()` functions.
```
Comment (ruisebas) [inline src/pages/[platform]/build-a-backend/data/set-up-data/index.mdx:958]:
```suggestion
First, add a private variable to store the subscription. Then create the subscription on the `init()` initializer, and add the `subscribe()` and `cancel()` functions.
```

File: src/pages/[platform]/build-a-backend/data/set-up-data/index.mdx [hunk 8]
```diff
@@ -444,20 +442,53 @@ com.example.MyAmplifyApp I/MyAmplifyApp: Added Todo with id: SOME_TODO_ID
...
 <InlineFilter filters={["swift"]}>
 
-Go to your **ContentView.swift** and add a button to create a new todo:
+Create a new file called `TodoViewModel.swift` and the `createTodo` function with the following code:
 
-```swift title="ContentView.swift"
-import SwiftUI
-// highlight-next-line
+```swift title="TodoViewModel.swift"
 import Amplify
```
Comment (lawmicha) [inline src/pages/[platform]/build-a-backend/data/set-up-data/index.mdx:448]:
```suggestion
import Foundation
import Amplify
```

File: src/pages/[platform]/build-a-backend/data/set-up-data/index.mdx (hunk 6 above)
Comment (lawmicha) [reply src/pages/[platform]/build-a-backend/data/set-up-data/index.mdx:452]: thanks!

**Review by ruisebas — APPROVED** [review]

**Review by renebrandel — APPROVED** [review]

**Pull request merged**